r/R/dplyr-funcs-datetime.R

```diff
@@ -463,3 +463,54 @@ duration_from_chunks <- function(chunks) {
   }
   duration
 }
+
+binding_as_date <- function(x,
+                            format = NULL,
+                            tryFormats = "%Y-%m-%d",
+                            origin = "1970-01-01",
+                            use_tz = "UTC",
+                            base = TRUE) {
```

Comment:
Sorry if I was unclear on this: I meant that we should rename `base` to `use_tz` and keep `tz` like it is. Though even still, like I said the first time: I suspect we don't actually need this `base` argument at all, and instead we can alter `x` to include the TZ before calling `binding_as_date` and then not carry around this extra argument: 

basically, you make the `as.Date` binding be:

```
if (call_binding("is.POSIXct", x)) {
      x <- build_expr("cast", x, options = cast_options(to_type = timestamp(timezone = tz)))
}
binding_as_date(x, ...)
```

But if the argument is necessary, let's call it something that is explicit about what it does.

Reply:
I see what you mean now. I have a new variant in which `binding_as_date()` no longer needs a `tz` argument since `tz` is used outside it (inside the binding definition). I kinda like this approach as it brings front and center the difference between `base::as.Date()` and `lubridate::as_date()`.

Reply:
Done